src/Hcal/HcalDigiProducer.cxx

```diff
@@ -95,27 +145,224 @@ void HcalDigiProducer::produce(framework::Event& event) {
+      {
+        adjustedYield=randGaussQ_.fire(scintillationYield_, scintillationYield_*scintillationYieldSigma_);
+      } while(adjustedYield<scintillationYield_*scintillationYieldCutoffLow_ ||
+              adjustedYield>scintillationYield_*scintillationYieldCutoffHigh_);
+
+      currentScintillationYield = scintillationYieldsAdjusted_.emplace(detID,adjustedYield).first;
+    }
+    photonGenerator->second->SetScintillationYield(currentScintillationYield->second);
+
+    CLHEP::Hep3Vector pos1Local, pos2Local;
+    for(int i=0; i<3; ++i)
+    {
+      pos1Local[i]=simHit.getPreStepPosition().at(i);
+      pos2Local[i]=simHit.getPostStepPosition().at(i);
+  if(std::fabs(pos1Local[0])>10.0) throw std::runtime_error("step point bigger than lookup table thickness");
+  if(std::fabs(pos2Local[0])>10.0) throw std::runtime_error("step point bigger than lookup table thickness");
+  if(std::fabs(pos1Local[1])>25.0) throw std::runtime_error("step point bigger than lookup table width");
+  if(std::fabs(pos2Local[1])>25.0) throw std::runtime_error("step point bigger than lookup table width");
+  if(std::fabs(pos1Local[2])>length/2.0) throw std::runtime_error("step point bigger than lookup table length");
+  if(std::fabs(pos2Local[2])>length/2.0) throw std::runtime_error("step point bigger than lookup table length");
+    }
+
+    int nContributions = simHit.getNumberOfContribs();
+    if(nContributions==0) continue;
+    int trackID = simHit.getContrib(0).trackID;  //There should only be 1 contributions for Hcal hits
+    double charge = particleMap[trackID].getCharge();
+
+    int absorber=0;  //no absorber at back section
+    if(section!=ldmx::HcalID::HcalSection::BACK)
+    {
+      if(layer%2==1) absorber=2; //for odd layer numbers at side sections, absorber is at positive end   //TODO: Is this true?
+      else
+      {
+        switch (section)
+        {
+          case ldmx::HcalID::HcalSection::TOP    : 
+          case ldmx::HcalID::HcalSection::RIGHT  : absorber=2;  //absorber at positive end
+                                                   break;
+          case ldmx::HcalID::HcalSection::BOTTOM : 
+          case ldmx::HcalID::HcalSection::LEFT   : absorber=-2;  //absorber at negative end
+                                                   break;
+        }
+      }
+    }
+
+    photonGenerator->second->MakePhotons(pos1Local, pos2Local, simHit.getPreStepTime(), simHit.getPostStepTime(),
+                                         simHit.getVelocity()/CLHEP::c_light, charge,
+                                         simHit.getEdep(),
+                                         simHit.getPathLength(),
+                                         absorber);
+
+
+    const std::vector<double> &timesPosEnd=photonGenerator->second->GetArrivalTimes(1);  //SiPM1 is at the positive and in the photonGenerator
+    const std::vector<double> &timesNegEnd=photonGenerator->second->GetArrivalTimes(0);  //SiPM0 is at the negative end in the photonGenerator
```

Comment:
@ehrlich-uva What shall we interpret as the positive and negative end in the photonGenerator? Does the photonGenerator know about the global geometry and whether positive end means +X/+Y?

Reply:
In this code, I define the positive sides to be at +x/+y/+z and set SiPM 1 at the positive side of a counter. Based on this, I decide at lines 215 ... 231 at which side the absorber is located at the counter (-2 at the negative side and +2 at the positive side). The photon generator uses this information to decide whether SiPM 0 or SiPM 1 gets read out.